amr-wind/immersed_boundary/complex_terrain/flat_ops.H

```diff
@@ -72,6 +78,66 @@ struct InitDataOp<Flat>
     }
 };
 
+template <>
+struct ComputeSourceOp<Flat>
```

Comment:
We can write this in a general form so it is applied to all the complex terrain ops and doesn't have to be re-written every time through templates. That would look like this:

```c++
template <typename IbType, std::enable_if_t<std::is_base_of<ComplexTerrainType, IbType>::value>>
```
You would need to move this to `IBOps.H` and adjust the data types to be based off your template type, but it would make it so we only had to write this function once.  
A couple of examples of this can be found in the ActuatorDisk PR I'm working on now: https://github.com/Exawind/amr-wind/pull/583